tiflash/tiflash-configuration.md

```diff
@@ -35,25 +35,65 @@ This section introduces the configuration parameters of TiFlash.
 
 ```toml
-tmp_path = The path in which the TiFlash temporary files are stored.
-path = The TiFlash data storage path.     # If there are multiple directories, separate each directory with a comma.
-path_realtime_mode = false # The default value is `false`. If you set it to `true` and multiple directories are deployed in the path, the latest data is stored in the first directory and older data is stored in the rest directories.
-listen_host = The listening host for supporting services such as TPC/HTTP. It is recommended to configure it as `0.0.0.0`.
-tcp_port = The TiFlash TCP service port.
-http_port = The TiFlash HTTP service port.
-mark_cache_size = 5368709120 # The cache size limit of the metadata of a data block. Generally, you do not need to change this value.
-minmax_index_cache_size = 5368709120 # The cache size limit of the min-max index of a data block. Generally, you do not need to change this value.
-```
+## The listening host for supporting services such as TPC/HTTP. It is recommended to configure it as "0.0.0.0", which means to listen on all IP addresses of this machine.
+listen_host = "0.0.0.0"
+## The TiFlash TCP service port.
+tcp_port = 9000
+## The TiFlash HTTP service port.
+http_port = 8123
+## The cache size limit of the metadata of a data block. Generally, you do not need to change this value.
+mark_cache_size = 5368709120
+## The cache size limit of the min-max index of a data block. Generally, you do not need to change this value.
+minmax_index_cache_size = 5368709120
+
+## The storage path of TiFlash data. If there are multiple directories, separate each directory with a comma.
+## `path` and `path_realtime_mode` are deprecated since v4.0.9. Use the configurations
+## in the `[storage]` section to get better performance in the multi-disk deployment scenarios
+# path = "/tidb-data/tiflash-9000"
+## or
+# path = "/ssd0/tidb-data/tiflash,/ssd1/tidb-data/tiflash,/ssd2/tidb-data/tiflash"
+## The default value is `false`. If you set it to `true` and multiple directories 
+## are set in the path, the latest data is stored in the first directory and older
+## data is stored in the rest directories.
+# path_realtime_mode = false 
+
+## The path in which the TiFlash temporary files are stored. By default it is the first directory in `path`
+## or `storage.latest.dir` appended with "/tmp"
+# tmp_path = "/tidb-data/tiflash-9000/tmp"
+
+## Storage paths settings effective since v4.0.9
+[storage]
+    [storage.main]
+    ## The directory(ies) to store the main data. More than 90% of the total data is stored in
+    ## these directories.
+    dir = [ "/tidb-data/tiflash-9000" ] 
+    ## or
+    # dir = [ "/ssd0/tidb-data/tiflash", "/ssd1/tidb-data/tiflash" ]
+
+    ## The maximum storage capacity of each directory in `storage.main.dir`.
+    ## If it is not set, or is set to multiple 0, the actual disk (the disk where the directory is located) capacity is used.
+    ## Note that big numbers such as "10GB" are not supported yet.
```

Comment:
This is for those users who see TiKV configuration. They may think that TiFlash supports this definition of a human-readable number too. But actually, TiFlash does not.
```suggestion
    ## Note that human-readable numbers such as "10GB" are not supported yet.
```